src/sql-parser/src/parser.rs

```diff
@@ -1635,8 +1635,63 @@ impl<'a> Parser<'a> {
         }))
     }
 
-    fn parse_connector(&mut self) -> Result<Connector, ParserError> {
-        match self.expect_one_of_keywords(&[FILE, KAFKA, KINESIS, AVRO, S3])? {
+    fn parse_connector(&mut self) -> Result<Connector<Raw>, ParserError> {
+        match self.expect_one_of_keywords(&[FILE, KAFKA, KINESIS, AVRO, S3, POSTGRES])? {
+            POSTGRES => {
+                self.expect_keyword(HOST)?;
+                let conn = self.parse_literal_string()?;
+                self.expect_keyword(PUBLICATION)?;
+                let publication = self.parse_literal_string()?;
+                self.expect_keyword(NAMESPACE)?;
+                let namespace = self.parse_literal_string()?;
+                self.expect_keyword(TABLE)?;
+                let table = self.parse_literal_string()?;
+
+                let mut columns = vec![];
+                self.expect_token(&Token::LParen)?;
+                if !self.consume_token(&Token::RParen) {
+                    loop {
```

Comment:
@mjibson moving review of the parser to this PR. Tables can have zero columns and `parse_comma_separated` assumes at least one item. I think this is why we parse columns in `CREATE TABLE` the same way (this is where I got the logic from).

Reply:
Ah, then could we just use `parse_columns` here? It's ok if that makes the syntax a bit more permissive if we can just reject some stuff in the planner; I noticed you already have the code to reject constraints besides `NOT NULL`.

Reply:
:+1: I changed the logic to use `parse_columns` and added one more check during planning!